### ISSUE TITLE
Can't send Chinese message !

### ISSUE THREAD

**Issue by chy168** [emesene/emesene#142]
Just try send message "Hello 台灣! (F)" with Chinese word.

We can got error:

```
[02:24:50 WARNING e3.common.Signal] Signal handler (_on_message_send_failed._on_message_send_failed) error: 'bool' object has no attribute 'keys'
Traceback (most recent call last):
  File "/home/chy168/emesene-svn/developing/emesene2-chy168/emesene/e3/common/Signal.py", line 63, in emit
    if callback(*args, **kwargs) == False:
  File "/home/chy168/emesene-svn/developing/emesene2-    chy168/emesene/e3/common/Signal.py", line 110, in __call__
    apply(self.f, (self.c(),) + arg)
  File "/home/chy168/emesene-svn/developing/emesene2-    chy168/emesene/gui/base/ConversationManager.py", line 80, in _on_message_send_failed
conversation.output.append(error,self.session.config.b_allow_auto_scroll)
  File "/home/chy168/emesene-svn/developing/emesene2-chy168/emesene/gui/gtkui/TextBox.py", line 273, in append
text = MarkupParser.parse_emotes(text, cedict)
  File "/home/chy168/emesene-svn/developing/emesene2-    chy168/emesene/gui/base/MarkupParser.py", line 35, in parse_emotes
 shortcuts.extend(cedict.keys())
```

  AttributeError: 'bool' object has no attribute 'keys'

**Comment by chy168** [comment]
In case, the message won't be sent.
I fixed this problem by encode body string to Unicode hex first.

chy168/emesene@8ce1291912710a2e1ace24a79e9327c986e8e3e1

Now, I can send the Chinese !!!

But, the AttributeError is still exist, i think.
Maybe that is "cedict" problem ?

**Comment by GVisconti84** [comment]
Uhm, at least in Qt4 frontend I'm able to send Japanese text and your text too....
Could you check what happens without your patch, and using papyon instead of e3.MSN? (Looking your patch seems like you're using e3.MSN, right?)

Exception happens in gui code, so I suppose that there are problems in how the gtk gui handles unicode.

Do you have custom emoticons saved?
I don't think the problem is in parse_emotes 'cause here it works with your string, but I see that your exception happens while handling custom emoticons: It could be a parse_emotes' bug that occurs only with unicode+custom emoticons

**Comment by chy168** [comment]
Yes, I use e3.MSN. (How shoud I do for choosing papyon ?)

I've trace the code from GTK conversation to protocol. The "AttributeError" is happened after sending Chinese word. Exactly, I can sent "嗨" and GUI will append "嗨" to Conversation window successfully. But my friend cannot receive "嗨" ...

Did Qt4 frontend handle the hex unicode with instance message ?

I don't have custom emoticons, the Chinese problem happened even just only one word.

I think, this problem is split two, one is MSN protocol problem, another is "cedict" at _on_message_send_failed. (1st problem trigger 2nd).

**Comment by c10ud** [comment]
you get the error because sending message fails -- please open a new issue for the e3.msn bug (and please use papyon instead... git submodule init && git submodule update)

**Comment by c10ud** [comment]
Fix append conversation errors in output. Closed by 340903b90be55f56f47b5317e6959a748df9b0d3
